src/Parser/Core/Modules/StatTracker.js

```diff
@@ -0,0 +1,257 @@
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      haste: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+
+    //// MISC ////
+    [SPELLS.JACINS_RUSE.id]: { mastery: 3000 },
+
+    //// DRUID ////
+    [SPELLS.ASTRAL_HARMONY.id]: { mastery: 4000 },
+
+
+  }
+
+  _startingStats = {};
+  _stats = {};
+
+  on_initialized() {
+    this._startingStats = {
+      str: this.combatants.selected.strength,
+      agi: this.combatants.selected.agility,
+      int: this.combatants.selected.intellect,
+      crit: this.combatants.selected.critRating,
+      haste: this.combatants.selected.hasteRating,
+      mastery: this.combatants.selected.masteryRating,
+      vers: this.combatants.selected.versatilityRating,
+    };
+    this._stats = this._startingStats;
+
+    debug && this._debugPrintStats(this._stats);
+  }
+
+  /*
+   * Stat rating at pull.
+   * Should be identical to what you get from Combatant.
+   */
+  get startingStrengthRating() {
+    return this._startingStats.str;
+  }
+  get startingAgilityRating() {
+    return this._startingStats.agi;
+  }
+  get startingIntellectRating() {
+    return this._startingStats.int;
+  }
+  get startingCritRating() {
+    return this._startingStats.crit;
+  }
+  get startingHasteRating() {
+    return this._startingStats.haste;
+  }
+  get startingMasteryRating() {
+    return this._startingStats.mastery;
+  }
+  get startingVersRating() {
+    return this._startingStats.vers;
+  }
+
+  /*
+   * Current stat rating, as tracked by this module.
+   */
+  get currentStrengthRating() {
+    return this._stats.str;
+  }
+  get currentAgilityRating() {
+    return this._stats.agi;
+  }
+  get currentIntellectRating() {
+    return this._stats.int;
+  }
+  get currentCritRating() {
+    return this._stats.crit;
+  }
+  get currentHasteRating() {
+    return this._stats.haste;
+  }
+  get currentMasteryRating() {
+    return this._stats.mastery;
+  }
+  get currentVersRating() {
+    return this._stats.vers;
+  }
+
+  /*
+   * For percentage stats, the percentage you'd have with zero rating.
+   * These values don't change.
+   */
+  get baseCritPercentage() {
+    return 0.08;
+  }
+  get baseHastePercentage() {
+    return 0;
+  }
+  get baseMasteryPercentage() {
+    switch (this.combatants.selected.spec) {
+      case SPECS.HOLY_PALADIN:
+        return 0.12;
+      case SPECS.HOLY_PRIEST:
+        return 0.05;
+      case SPECS.RESTORATION_SHAMAN:
+        return 0.24;
+      case SPECS.ENHANCEMENT_SHAMAN:
+        return 0.2;
+	    case SPECS.RESTORATION_DRUID:
+	      return 0.048;
+      default:
+        throw new Error('Mastery hasn\'t been implemented for this spec yet.');
+    }
+  }
+  get baseVersPercentage() {
+    return 0;
+  }
+
+  /*
+   * For percentage stats, this is the multiplier to go from rating to percent (expressed from 0 to 1)
+   * These values don't change.
+   */
+  get critRatingToPercent() {
+    return (1 / 40000);
```

Comment:
Parens are unnecessary and imo distracting.

Reply:
I'll get rid of them.